biz.aQute.bndlib/src/aQute/bnd/cdi/AnnotationReader.java

```diff
@@ -0,0 +1,340 @@
+package aQute.bnd.cdi;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.annotation.ElementType;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.osgi.service.cdi.annotations.Bean;
+import org.osgi.service.cdi.annotations.Beans;
+import org.osgi.service.cdi.annotations.MinimumCardinality;
+import org.osgi.service.cdi.annotations.Reference;
+import org.osgi.service.cdi.annotations.Service;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+
+import aQute.bnd.cdi.CDIAnnotations.Options;
+import aQute.bnd.osgi.Analyzer;
+import aQute.bnd.osgi.Annotation;
+import aQute.bnd.osgi.ClassDataCollector;
+import aQute.bnd.osgi.Clazz;
+import aQute.bnd.osgi.Clazz.FieldDef;
+import aQute.bnd.osgi.Clazz.MethodDef;
+import aQute.bnd.osgi.Clazz.QUERY;
+import aQute.bnd.osgi.Descriptors.PackageRef;
+import aQute.bnd.osgi.Descriptors.TypeRef;
+import aQute.bnd.osgi.Instruction;
+import aQute.bnd.signatures.ClassTypeSignature;
+import aQute.bnd.signatures.FieldSignature;
+import aQute.bnd.signatures.JavaTypeSignature;
+import aQute.bnd.signatures.MethodSignature;
+import aQute.bnd.signatures.TypeArgument;
+import aQute.bnd.version.Version;
+
+public class AnnotationReader extends ClassDataCollector {
```

Comment:
AnnotationReader is a crappy name for the DS annotation reader class. Now we will have two such named classes. Better to name this one `CdiAnnotationReader` or something similiar.